### PR TITLE
Chore/multi language fixes

### LINKED ISSUE #179: Multi-language Fixes and Improvements
#### Description:
This issue tracks the tasks needed to enhance multi-language support and functionality in our application.

#### Tasks:
1. **Get lessons markdown inside lessons page**:
   - Retrieve lesson markdown content directly within the lessons page.

2. **Implement GetPage inside getServerSideProps instead of getLessons**:
   - Refactor code to use `GetPage` within `getServerSideProps` for better performance.

3. **Fix lesson submission validation**:
   - Correct issue where lesson submissions are not properly validated.

4. **Make repository language dynamic based on locale**:
   - Update repository language settings to dynamically change based on user locale.
   
5. **Update page path url to include Section number.**

#### Goal:
Ensure content is correctly localized and validated, enhancing the user experience across different languages.

### PR DESCRIPTION
**Pull request by nomadbitcoin**
This PR includes several key updates and fixes to improve multi-language support and functionality.

- Standardized language support for Brazilian Portuguese.
- Ensured content loads correctly based on the selected language and updated the URL path accordingly.
- Refactor code to use GetPage within getServerSideProps for better performance.
- Update page path url to include Section number.

Closes #179

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   i18n: {
-    locales: ['en', 'pt'],
+    locales: ['en', 'pt_BR'],
```
Comment (danicuki) [inline next.config.js:3]: The correct is pt-BR (with hiphen)

File: pages/courses/[id]/[section]/[lesson].js [hunk 2]
```diff
@@ -58,47 +58,70 @@ function Lessons({ course, lesson, currentDate }) {
...
+      setLessonsSubmitted(lessonsSubmitted_)
+    }
+    fetchLessonsSubmitted()
+  }, [user, cohort, open])
 
   useEffect(() => {
-    lessonsSubmitted.map((item) => {
-      if (item?.lesson === lesson && item?.user == user?.uid && item?.cohort_id == cohort?.id) {
+    lessonsSubmitted.forEach((item) => {
+      console.log('lesson', lesson)
```
Comment (danicuki) [inline pages/courses/[id]/[section]/[lesson].js:71]: Left over

File: pages/courses/[id]/[section]/[lesson].js [hunk 3]
```diff
@@ -58,47 +58,70 @@ function Lessons({ course, lesson, currentDate }) {
...
+    const fetchLessonsSubmitted = async () => {
+      let lessonsSubmitted_ = await getLessonsSubmissions(user?.uid, cohort?.id)
+      setLessonsSubmitted(lessonsSubmitted_)
+    }
+    fetchLessonsSubmitted()
+  }, [user, cohort, open])
 
   useEffect(() => {
-    lessonsSubmitted.map((item) => {
-      if (item?.lesson === lesson && item?.user == user?.uid && item?.cohort_id == cohort?.id) {
```
Comment (danicuki) [inline pages/courses/[id]/[section]/[lesson].js:67]: Why did you remove this check?

File: pages/courses/[id]/[section]/[lesson].js [hunk 4]
```diff
@@ -58,47 +58,70 @@ function Lessons({ course, lesson, currentDate }) {
...
+    i18n.on('languageChanged', handleLanguageChange)
+
+    return () => {
+      i18n.off('languageChanged', handleLanguageChange)
+    }
+  }, [i18n, router])
+
   const nextLesson = () => {
-    const currentLessonIndex = sortedLessons.map((item) => item.lesson === lesson).indexOf(true)
+    const currentLessonIndex = sortedLessons.findIndex((item) => item.lesson === lesson)
```
Comment (danicuki) [inline pages/courses/[id]/[section]/[lesson].js:101]: 😍

File: pages/courses/[id]/[section]/[lesson].js [hunk 5]
```diff
@@ -58,47 +58,70 @@ function Lessons({ course, lesson, currentDate }) {
...
       }
     })
-  }, [lessonsSubmitted])
+  }, [lessonsSubmitted, lesson])
 
   useEffect(() => {
     setSortedLessons(course.lessons.sort((a, b) => (a.section > b.section ? 1 : -1)))
   }, [sortedLessons])
+
+  useEffect(() => {
```
Comment (danicuki) [inline pages/courses/[id]/[section]/[lesson].js:83]: This code should not be here. It is a generic language change code that should be in the page template or navbar component.

**Review by danicuki — CHANGES_REQUESTED** [review]

File: next.config.js (hunk 1 above)
Comment (nomadbitcoin) [reply next.config.js:3]: Okay, I'll update that in the https://github.com/w3b3d3v/buildspace-projects as well

File: pages/courses/[id]/[section]/[lesson].js (hunk 3 above)
Comment (nomadbitcoin) [reply pages/courses/[id]/[section]/[lesson].js:67]: In the previous code, "lessonsSubmitted" included all users' lessons from other courses and cohorts. However, my solution only fetches the lessons for the current course and cohort directly in the database query, eliminating the need for additional checks.

File: pages/courses/[id]/[section]/[lesson].js (hunk 5 above)
Comment (nomadbitcoin) [reply pages/courses/[id]/[section]/[lesson].js]: Removed 😉

**Review by danicuki — APPROVED** [review]

**Pull request merged**